doc/source/io.rst

```diff
@@ -2801,6 +2801,24 @@ everything in the sub-store and BELOW, so be *careful*.
 
 .. _io.hdf5-types:
 
+.. warning:: Hierarchical keys cannot be retrieved as dotted (attribute) access as described above for items stored under root node.
```

Comment:
This belongs in a separate PR --- or I would consider renaming this one to make it obvious that this is the big change here.